.github/workflows/e2e.yaml

```diff
@@ -97,10 +104,18 @@ jobs:
       - id: generate-cluster-name
         name: generate cluster name
         run: |
-          CLUSTER_NAME="$(echo ${{ inputs.suite }}-"$RANDOM$RANDOM" | awk '{print tolower($0)}' | tr / -)"
-          echo Using cluster name "$CLUSTER_NAME"
+          CLUSTER_NAME=''
+          if [[ '${{ inputs.cluster_name }}' == '' ]] && [[ '${{ inputs.workflow_trigger }}' == 'soak' ]]; then 
```

Comment:
optional: maybe it's just me but I find this more confusing than the last iteration

Reply:
I agree. Sorry If I said otherwise, I honestly don't remember, but I think it would look better like the following

```
if cluster_name is unset {
   if soak { cluster_name = soak }
   else { cluster_name = not_soak }
} else {
  cluster_name is existing
}
```